### PR TITLE
refactor(`monitoringp`): improve code quality

### LINKED ISSUE #466: Improve quality of code generated from templates
Improve quality from generated code for the `monitoring-provider` module for the end of Milestone 3

Indication: https://github.com/tendermint/spn/issues/217

### PR DESCRIPTION
**Pull request by aljo242**
Closes #466

### What does this PR does?

- Basic fixes to naming and function signatures
- Refactors keeper tests to follow the `TestKeeper` structure introduced in #579
- Removes unnecessary simulation files and functions

### CODE REVIEW

File: x/monitoringp/genesis_test.go [hunk 1]
```diff
@@ -3,9 +3,10 @@ package monitoringp_test
 import (
 	"testing"
 
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
```
[the remark this answers is not in the document]
Comment (aljo242) [reply x/monitoringp/genesis_test.go]: Ah that was a mistake - thanks

**Review by Pantani — DISMISSED** [review]

File: testutil/keeper/monitoring_provider.go [hunk 2]
```diff
@@ -3,49 +3,103 @@ package keeper
...
+	participationtypes "github.com/tendermint/spn/x/participation/types"
+	profilekeeper "github.com/tendermint/spn/x/profile/keeper"
+	rewardkeeper "github.com/tendermint/spn/x/reward/keeper"
+	rewardtypes "github.com/tendermint/spn/x/reward/types"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	ibckeeper "github.com/cosmos/ibc-go/v2/modules/core/keeper"
 	"github.com/stretchr/testify/require"
 	"github.com/tendermint/tendermint/libs/log"
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
```
Comment (lubtd) [inline testutil/keeper/monitoring_provider.go:23]: We should put these above `spn` import

File: x/monitoringc/module_simulation.go [hunk 3]
```diff
@@ -18,7 +18,6 @@ import (
 // avoid unused import issue
 var (
 	_ = sample.AccAddress
-	_ = monitoringcsimulation.FindAccount
 	_ = simappparams.StakePerAccount
 	_ = simulation.MsgEntryKind
 	_ = baseapp.Paramspace
```
Comment (lubtd) [inline x/monitoringc/module_simulation.go:23]: Why we need those?
@Pantani do you have some clue?

File: x/monitoringp/simulation/simulation.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package simulation
```
Comment (lubtd) [inline x/monitoringp/simulation/simulation.go:1]: Why creating this file?

**Comment by lubtd** [comment]
I renamed it into `refactor` I think is more adapted since `chore` normally doesn't have influence in the codebase

File: x/monitoringp/simulation/simulation.go (hunk 4 above)
Comment (aljo242) [reply x/monitoringp/simulation/simulation.go:1]: Just to make things symmetric with the other packages.  At some point, I assume we will actually create the simulations, so this is a placeholder.

File: x/monitoringc/module_simulation.go (hunk 3 above)
Comment (aljo242) [reply x/monitoringc/module_simulation.go:23]: I'm not exactly sure actually.  Seems like I can remove all of them and can build / simulate everything fine.

Comment (aljo242) [reply x/monitoringc/module_simulation.go]: See #612

**Review by lubtd — APPROVED** [review]

**Pull request merged**